sdk/Sdk.Generators/FunctionExecutor/FunctionExecutorGenerator.ExecutableFunction.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Generators
+{
+    /// <summary>
+    /// A type which holds information about the functions which can be executed from an invocation.
+    /// </summary>
+    internal class ExecutableFunction
+    {
+        /// <summary>
+        ///  False if the function returns Task or void.
+        /// </summary>
+        internal bool IsReturnValueAssignable { set; get; }
+
+        /// <summary>
+        /// Whether the function should be awaited or not for getting the result of execution.
+        /// </summary>
+        internal bool ShouldAwait { get; set; }
+
+        /// <summary>
+        /// The method name (which is part of EntryPoint property value).
+        /// </summary>
+        internal string MethodName { get; set; } = null!;
+
+        internal bool IsStatic { get; set; }
```

Comment:
I felt the property name itself explains what it does and this type is not public.

Reply:
I definitely favour oversharing rather than under shareing :) The others are not public either and never hurts to have more information. 

```
/// <summary>
/// Gets or sets whether the function is static
/// </summary>